### PR TITLE
Feature/153/labeling packages in legend

### PR DESCRIPTION
**Pull request by alschmut**
# Feature/153/labeling packages in legend

## Description

solves #153 

The Legend now shows the package names and paths of the packages, which were assigned to a marking color. As an example an image of the legend:

- Hovering over the package name shows a tooltip-title with the full path
- The overall length of the shown path is set variably with [`MAX_NAME_LENGTH`](https://github.com/MaibornWolff/codecharta/blob/feature/153/labeling-packages-in-legend/visualization/app/codeCharta/ui/legendPanel/legendPanelComponent.ts#L166)
1. When the package name is short enough: a part of the package path is shown
2. When the package name is too long: it shows the first and last part with ... in between

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

* [x] **All** requirements mentioned in the issue are implemented
* [x] Does match the Code of Conduct and the Contribution file 
* [x] Task has its own **GitHub issue** (something it is solving)
  * [x] Issue number is **included in the commit messages** for traceability
* [x] **Update the README.md** with any changes/additions made
* [x] **Update the CHANGELOG.md** with any changes/additions made
* [ ] **Enough test coverage to ensure that uncovered changes do not break functionality**
* [ ] **All tests pass**    
* [x] **Descriptive pull request text**, answering:
  + What problem/issue are you fixing?
  + What does this PR implement and how? 
* [x] **Assign your PR to someone** for a code review
  + This person _will be contacted **first**_ if a bug is introduced into `master`
* [x] **Manual testing** did not fail

### CODE REVIEW

**Comment by alschmut** [comment]
@mrpatpat could you help me with the [legendPanelController.spec.js](https://github.com/MaibornWolff/codecharta/blob/feature/153/labeling-packages-in-legend/visualization/app/codeCharta/ui/legendPanel/legendPanelController.spec.js)? Its still written without jest and I don't get the old code updated. Then I could write new tests for the new functions.

**Comment by Richargh** [comment]
I like it! Can we create a bit of space between the building colors and the folder colors though? Just half a line or one line so to separate the different color dots.

**Comment by Richargh** [comment]
Sorry I made a mistake and closed this by mistake.

**Comment by mrpatpat** [comment]
I added some suggestions. It works fine and looks nice.

File: visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import {CodeMapNode} from "../../core/data/model/CodeMap";
 import {hierarchy} from "d3-hierarchy";
-import {SettingsService} from "../../core/settings/settings.service";
+import {MarkingPackages, SettingsService} from "../../core/settings/settings.service";
```
[the remark this answers is not in the document]
Comment (alschmut) [reply visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts]: Oh yes, I forgot to take out that import

File: visualization/app/codeCharta/core/settings/settings.service.ts [hunk 2]
```diff
@@ -22,6 +22,16 @@ export interface Scale {
     z: number;
 }
 
+export interface MarkingPackages {
```
[the remark this answers is not in the document]
Comment (alschmut) [reply visualization/app/codeCharta/core/settings/settings.service.ts]: yes true, thanks! I used to store them inside the settings, but I changed my mind afterwards.

**Pull request merged**